pkg/lightning/backend/local/local.go

```diff
@@ -217,21 +239,41 @@ func (e *File) getEngineFileSize() backend.EngineFileSize {
 	var memSize int64
 	e.localWriters.Range(func(k, v interface{}) bool {
 		w := k.(*Writer)
-		memSize += w.writeBatch.totalSize
 		if w.writer != nil {
-			total.Size += int64(w.writer.writer.EstimatedSize())
+			memSize += int64(w.writer.writer.EstimatedSize())
+		} else {
+			// if kvs are still in memory, only calculate half of the total size
+			// in our tests, SST file size is about 50% of the raw kv size
+			memSize += w.batchSize / 2
```

Comment:
Why half?

Reply:
The SST compaction ratio is about 2, so only calculated 50% of the raw size as disk-quota size